omod/src/main/java/org/openmrs/module/owa/web/controller/OwaRestController.java

```diff
@@ -102,126 +106,146 @@
 
 	@RequestMapping(value = "/rest/owa/addapp", method = RequestMethod.POST)
 	@ResponseBody
-	public List<App> upload(@RequestParam("file") MultipartFile file, HttpServletRequest request, HttpServletResponse response) throws IOException {
+	public ResponseEntity<List<App>>  upload(@RequestParam("file") MultipartFile file, HttpServletRequest request, HttpServletResponse response)
+	        throws IOException {
 		List<App> appList = new ArrayList<>();
-		if(Context.hasPrivilege("Manage OWA")){
-			String message;
+		if (Context.hasPrivilege("Manage OWA")) {
 			HttpSession session = request.getSession();
+			boolean checkInstall = false;
 			if (!file.isEmpty()) {
-				String fileName = file.getOriginalFilename();
-				File uploadedFile = new File(file.getOriginalFilename());
+				String passedFileName = file.getOriginalFilename();
+				String fileName = getFileName(passedFileName);
+				File uploadedFile = new File(fileName);
 				file.transferTo(uploadedFile);
-				try (ZipFile zip = new ZipFile(uploadedFile)) {
-					if (zip.size() == 0) {
-						message = messageSourceService.getMessage("owa.blank_zip");
-						uploadedFile.delete();
-						log.warn("Zip file is empty");
```

Comment:
Did you actually change all these? Or are they just formatting changes?

Reply:
since this action was being used both by install and upload. I separated it and created a method of it's own. This led to the multiple changes.

Reply:
Can you first leave it intact such that your only change are dealing with the bug fix? Then you can always later on create a new ticket for the refactoring. This way, it is easy to tell what exactly was changed for the fix alone.

Reply:
Okay

Reply:
I have finished updating the changes to only show those to fix the bug